core/git-access.c

```diff
@@ -434,10 +430,13 @@ static int try_to_git_merge(git_repository *repo, git_reference **local_p, git_r
 		goto write_error;
 	if (git_tree_lookup(&merged_tree, repo, &merge_oid))
 		goto write_error;
-	if (git_signature_default(&author, repo) < 0)
-		if (git_signature_now(&author, "Subsurface", "noemail@given") < 0)
-			goto write_error;
-	if (git_commit_create_v(&commit_oid, repo, NULL, author, author, NULL, "automatic merge", merged_tree, 2, local_commit, remote_commit))
+	if (get_authorship(repo, &author) < 0)
+		goto write_error;
```

Comment:
I'm surprised that in C you can jump across the definition of a variable. In C++ that wouldn't be accepted, because it would mean that you destroy the object without having initialized it.

That raises the question: what happens if the code jumps to `write_error` here? Will `free_buffer` be called on an unitialized `msg`?

Reply:
yes, and that would be BAD. I need to move the declaration / initialization to the top of the function.